tests/rptest/tests/vcluster_topic_property_test.py

```diff
@@ -0,0 +1,78 @@
+# by the Apache License, Version 2.0
+
+import random
+from rptest.clients.rpk import RpkException, RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from random import randbytes
+
+BASE32_ALPHABET = "0123456789abcdefghijklmnopqrstuv"
+
+
+def get_vcluster_id(topic_properties: dict):
+    if TopicSpec.PROPERTY_VIRTUAL_CLUSTER_ID not in topic_properties:
+        return None
+    return topic_properties[TopicSpec.PROPERTY_VIRTUAL_CLUSTER_ID][0]
+
+
+class VirtualClusterTopicPropertyTest(RedpandaTest):
+    def __init__(self, test_context):
+        super(VirtualClusterTopicPropertyTest,
+              self).__init__(test_context=test_context, num_brokers=3)
+
+    def _random_xid_string(self):
+        xid = ''.join([random.choice(BASE32_ALPHABET) for _ in range(19)])
+
+        return xid + random.choice(['0', 'g'])
+
+    @cluster(num_nodes=3)
+    def test_basic_virtual_cluster_property(self):
+        rpk = RpkTool(self.redpanda)
+        # try creating vcluster enabled topic with extensions disabled
+        topic = TopicSpec()
+        vcluster_id = self._random_xid_string()
+        rpk.create_topic(
+            topic=topic.name,
+            partitions=1,
+            replicas=3,
+            config={TopicSpec.PROPERTY_VIRTUAL_CLUSTER_ID: vcluster_id})
+
+        configs = rpk.describe_topic_configs(topic.name)
+        assert TopicSpec.PROPERTY_VIRTUAL_CLUSTER_ID not in configs, \
+            "When MPX extensions are disabled there should be no virtual cluster property reported"
+        try:
+            configs = rpk.alter_topic_config(
+                topic.name, TopicSpec.PROPERTY_VIRTUAL_CLUSTER_ID, vcluster_id)
+            assert False, "Altering topic virtual cluster property should not be supported"
+        except RpkException as e:
+            assert "INVALID_CONFIG" in e.msg
+        rpk.delete_topic(topic.name)
+        # enable mpx extensions
+        self.redpanda.set_cluster_config({"enable_mpx_extensions": True})
+        # create topic with virtual cluster id assigned
+        rpk.create_topic(
+            topic=topic.name,
+            partitions=1,
+            replicas=3,
+            config={TopicSpec.PROPERTY_VIRTUAL_CLUSTER_ID: vcluster_id})
+
+        configs = rpk.describe_topic_configs(topic.name)
+
+        assert get_vcluster_id(configs) == vcluster_id, \
+            f"current virtual cluster id {get_vcluster_id(configs)} and expected cluster id {vcluster_id} doesn't match"
+
+        try:
+            rpk.alter_topic_config(topic.name,
+                                   TopicSpec.PROPERTY_VIRTUAL_CLUSTER_ID,
+                                   self._random_xid_string())
+            assert False, "Altering topic virtual cluster property should not be supported"
+        except RpkException as e:
+            assert "INVALID_CONFIG" in e.msg
```

Comment:
does this need to be a dedicated test? we have tests for alter_topic_config already and this test seems to check that alter_topic_config is working correctly. maybe it doesn't matter if the idea is that we'll expand this with bigger and richer tests which need to set the cluster id?

Reply:
This test is not validating the general alter config behavior but rather the special smentics of the vcluster property it validates that the property is not alterable (it can only be set while creating the topic) and it is not setable when mpx extensions are disabled